### DISCUSSION TITLE
One to Many WiFi stream

### DISCUSSION THREAD

**Discussion by rmick** [pschatzmann/arduino-audio-tools#209] [Ideas]
I am working a project where I want to do two-way radio style comms and your library is damn near perfect.

The i2S to TCP/IP works perfectly for point to point (1 to 1) communications, but I would like a one to many scenario, where one device can send out a message and all the other devices on that 'channel' can hear it. The current implementation does not allow this as it only sends to a single IP address.

Is this possible.

here are a couple of ideas
1) Use multicast.
2) allow multiple clients to connect (but I think this breaks the Arduino client class).

**Comment by pschatzmann** [comment]
I started to work on the topic of streaming of audio data as well and suggest to use
- [ESP-NOW](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_e_s_p_now_stream.html)
- [UDP](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_u_d_p_stream.html)

I would also use it together with a [CODEC](https://www.pschatzmann.ch/home/2022/04/28/arduino-low-latency-streaming-of-audio-data-codecs/)

**Comment by bikelander** [comment]
Hello everyone, Mr Schatzmann,
Has there been any progress in this development?
Using "streams-i2s-webserver_wav" as a reference, I´m looking for a way that allows several clients, through their browsers, to listen to the audio broadcast transmission of one ESP Wifi sender. Any suggestion on how to accomplish that?
Thanks.

**Reply by pschatzmann** [reply]
I added the simple webserver just because I was getting tired of wiring up different modules to the processor and it was a convinient way to to test listen to the result. Even before the audio tools, I started to work on a [(real) http server](https://github.com/pschatzmann/TinyHttp), but I never had the chance to complete it.  I think it should not take too much effort to get this completed..

Here is the [example sketch](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-webserver/streams-generator-webserverex_wav)

Of cause the other suggested approaches will work as well...

**Reply by pschatzmann** [reply]
You can also use the [MultiOutput](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_multi_output.html) to send the audio to multiple destinations
